app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/SessionUpdates.scala

```diff
@@ -27,6 +27,8 @@ import scala.concurrent.Future
 
 trait SessionUpdates {
 
+  type SessionDataTransform = SessionData => SessionData
```

Comment:
Depending on type of work you trying to achieve here, perhaps I am not fully understand.
But state transition I would recommend using `State` from `Cats` glossary.
https://typelevel.org/cats/datatypes/state.html

Reply:
There is no state involved here, it's just a shorthand.